packages/embark/src/lib/modules/tests/index.js

```diff
@@ -228,13 +238,16 @@ class TestRunner {
         return cb(err);
       }
       let failures = runs.reduce((acc, val) => acc + val, 0);
-      self.fs.remove('.embark/contracts', (_err) => {
-        cb(null, {failures});
+      self.events.request('config:contractsFiles:reset', () => {
+        global.config({}, (err) => {
+          cb(err, {failures});
+        });
       });
     });
   }
 
   runSolidityTests(files, options, cb) {
+    const self = this;
```

Comment:
Considering that we're inside class methods and `self` is used within fat arrow functions, I'd expect this to work with just `this`.

@emizzle would you mind quadruple checking whether we really need to re-introduce `const self = this;` here?

Reply:
You're 100% correct, `this` is fine in this scope. I think it was more being lazy as the other code was copied.